### PR TITLE
Remove reference to parflowio, relax requirement on numba

### PR DESCRIPTION
**Pull request by arbennett**
This PR fixes a leftover reference to parflowio that wasn't removed in #365. It also makes the numba dependency optional as there are still some issues with installing on Mac M1 based machines.

### CODE REVIEW

File: pftools/python/setup.py [hunk 1]
```diff
@@ -23,20 +23,27 @@
     packages=find_packages(),
     install_requires=[
         'pyyaml==5.4',
```
Comment (jourdain) [inline pftools/python/setup.py:25]: Maybe relax that constraint to be >=5.4

File: pftools/python/setup.py [hunk 2]
```diff
@@ -23,20 +23,27 @@
...
-        'numba',
-        'dask',
-        'imageio>=2.9.0'
     ],
     include_package_data=True,
     extras_require={
         'all': [
-            'imageio>=2.9.0',
+            'imageio>=2.9.0'
+            'numpy',
```
Comment (jourdain) [inline pftools/python/setup.py:31]: is numpy core? or can it be truly optional? I don't know, I'm just wondering.

**Review by jourdain — APPROVED** [review]
Your change looks great! Thanks for making them. Might be worth doing 2 edits.
1) Update the version in setup.py so we can publish a new one.
2) Relax the yaml version constraint

File: pftools/python/setup.py (hunk 2 above)
Comment (arbennett) [reply pftools/python/setup.py:31]: I also don't know. I think this is how it was previously.

**Review by jourdain — APPROVED** [review]
LGTM but make sure you rebase on master. As I think you are slightly off master.

**Comment by jourdain** [comment]
If everything comeback green we should be able to merge. 

@callachennault do you want to try to publish the new version to PyPI just to make sure you are all-set on your side?

**Review by smithsg84 — APPROVED** [review]

**Comment by callachennault** [comment]
> If everything comeback green we should be able to merge.
> 
> @callachennault do you want to try to publish the new version to PyPI just to make sure you are all-set on your side?

@jourdain I wasn't able to get it working. I don't know if I'm missing a prerequisite to the steps [here](https://github.com/parflow/parflow/tree/master/pftools/python#how-to-update-this-package-developers-only).

**Comment by jourdain** [comment]
I just forwarded you an email explaining what I had to do. Hopefully that should help you.

**Pull request merged**